specification/ingest/simulate/types.ts

```diff
@@ -42,11 +43,12 @@ export class Document {
   _source: UserDefinedValue
 }
 
-export class DocumentSimulation {
-  _id: Id
-  _index: IndexName
+/**
+ * The simulated document, with optional metadata.
+ */
+export class DocumentSimulation
+  implements AdditionalProperties<string, string>
```

Comment:
I'm curious about this change, aren't the `_id`, `_index`, etc fields well-defined for this object?

Reply:
I looked at the ES source code and the only fields that are defined there are `_ingest` and `_source`. Everything else is a string->string metadata dictionary. So there's actually no guarantee that `_id` and `_index` will always be there, even if they happen to always be present in our recordings.

What triggered me to look at the code is that there was a missing `_version` field that is a number in a string. So I wanted to understand why it wasn't a real number and discovered this metadata dictionary 😉

Reply:
Interesting! The implementation shouldn't dictate the API contract though IMO. If certain fields like `_id` and `_index` are always there I think it makes sense to continue having at least those two fields in our API spec?